src/support.js

```diff
@@ -671,18 +741,25 @@ function addFunction(func, sig) {
 #endif
 
 #endif // EMULATED_FUNCTION_POINTERS == 0
+
+#endif // WASM_BACKEND
 }
 
 function removeFunction(index) {
+#if WASM_BACKEND
+#else
 #if EMULATED_FUNCTION_POINTERS == 0
   functionPointers[index-jsCallStartIndex] = null;
 #else
+#if !WASM
   alignFunctionTables(); // XXX we should rely on this being an invariant
   var tables = getFunctionTables();
   for (var sig in tables) {
     tables[sig][index] = null;
   }
 #endif
+#endif
+#endif
```

Comment:
comments to match these endifs with their ifs might be nice, if not early-return